src/app/screen/onboarding/restaurant-added-popup/restaurant-added-popup.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Location } from '@angular/common';
+import { RestaurantAddRequest } from 'src/app/models/RestaurantAddRequest';
+import { ActivatedRoute, Router } from '@angular/router';
+@Component({
+  selector: 'app-restaurant-added-popup',
+  templateUrl: './restaurant-added-popup.component.html',
+  styleUrls: ['./restaurant-added-popup.component.scss'],
+})
+export class RestaurantAddedPopupComponent {
+  handleAccept() {
+    // this.location.back();
+    this.router.navigate(['/all-restaurants']);
+  }
+  constructor(
+    private location: Location,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {}
+
+  // -------------------------------------------------------------------------
+  // one way to send data from one component to another  using queryParams
+
+  restaurantDetails: RestaurantAddRequest = new RestaurantAddRequest();
+
+  ngOnInit(): void {
+    this.route.queryParams.subscribe((params) => {
+      const serializedData = params['restaurantAddRequestData'];
+      if (serializedData) {
+        this.restaurantDetails = JSON.parse(serializedData);
```

Comment:
why all this complexity? what was the issue

Reply:
i got data in string format  from other component and now i want json object, so  now i have to convert it into json object and then i can use it.

Reply:
now , i have removed that part and added using @Input and child component